http-agent/src/test/java/org/motechproject/http/agent/listener/HttpClientEventListenerTest.java

```diff
@@ -197,4 +198,35 @@ public void shouldReturnResponse() throws Throwable {
         ResponseEntity<?> responseEntity = httpClientEventListener.handleWithUserPasswordAndReturnType(motechEvent);
         assertEquals(responseEntity, exceptedResponseEntity);
     }
+
+    @Test
+    public void shouldUseBasicRestTemplate() {

```

Comment:
add condition to the method name, like "when username and password are not provided"